IPython/core/page.py

```diff
@@ -208,9 +210,13 @@ def pager_page(strng, start=0, screen_lines=0, pager_cmd=None):
         else:
             try:
                 retval = None
-                # if I use popen4, things hang. No idea why.
-                #pager,shell_out = os.popen4(pager_cmd)
-                pager = os.popen(pager_cmd, 'w')
+                # Emulate os.popen, but redirect stderr
+                proc = subprocess.Popen(pager_cmd,
+                                shell=True,
+                                stdin=subprocess.PIPE,
+                                stderr=subprocess.DEVNULL
+                                )
+                pager = os._wrap_close(io.TextIOWrapper(proc.stdin), proc)
```

Comment:
Should we really use a function that is not part of the public API? What is the intention here?

Reply:
In this case I believe this is ok, as this is exactly what `os.popen` call. There is not much reason for it to change.